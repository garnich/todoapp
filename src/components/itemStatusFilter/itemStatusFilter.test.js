import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ItemStatusFilter from './itemStatusFilter'

Enzyme.configure({ adapter: new Adapter() })

describe('Test ItemStatusFilter component', () => {
  const props = {
    filter: 'all',
    onFilterChange: jest.fn(),
  }

  it('<ItemStatusFilter>', () => {
    const wrapper = shallow(<ItemStatusFilter {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('<ItemStatusFilter> simulate click', () => {
    const wrapper = shallow(<ItemStatusFilter {...props} />)
    wrapper
      .find('.btn-group')
      .children()
      .at(0)
      .simulate('click')

    expect(props.onFilterChange).toHaveBeenCalled()
  })
})
