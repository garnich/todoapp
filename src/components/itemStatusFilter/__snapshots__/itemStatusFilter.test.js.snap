// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ItemStatusFilter component <ItemStatusFilter> 1`] = `
<div
  className="btn-group"
>
  <button
    className="btn btn-info"
    key="all"
    onClick={[Function]}
    type="button"
  >
    All
  </button>
  <button
    className="btn btn-outline-secondary"
    key="todo"
    onClick={[Function]}
    type="button"
  >
    ToDo
  </button>
  <button
    className="btn btn-outline-secondary"
    key="done"
    onClick={[Function]}
    type="button"
  >
    Done
  </button>
</div>
`;
